app/lib/actions/katello/host/update_system_purpose.rb

```diff
@@ -0,0 +1,31 @@
+module Actions
+  module Katello
+    module Host
+      class UpdateSystemPurpose < Actions::EntryAction
+        def plan(host, service_level, purpose_role, purpose_usage, purpose_addons)
+          fail _("Host %s has not been registered with subscription-manager.") % host.name unless host.subscription_facet
+
+          host.subscription_facet.service_level = service_level unless service_level.nil?
+          host.subscription_facet.purpose_role = purpose_role unless purpose_role.nil?
+          host.subscription_facet.purpose_usage = purpose_usage unless purpose_usage.nil?
+
+          if purpose_addons
```

Comment:
Do the tests still pass if this is consolidated:

```
if purpose_addons
  purpose_addon_objects = purpose_addons.reject(&:blank?).uniq.map.....
  host.subscription_facet.purpose_addons = purpose_addon_objects
end

Reply:
Yeah they do, will clean it up to that +1

Reply:
Pushed changes